### PR TITLE
Workaround performance/timeout issue for 'vnc-two-passwords' on aarch64

### PR DESCRIPTION
**Pull request by Zaoliang**
we experience sporadic issue with assert_screen 'tigervnc-desktop-loggedin',
so increase timeout value can be a workaround on some weak workers.
and further workarounds provided as welll for xev and performance issue.
xev hangs sporadic (1%) which cannot be fixed at moment, so provide record_info with ticket
poo#120282 for a later solution
see https://progress.opensuse.org/issues/119416
VRs:
https://openqa.suse.de/tests/9931082
https://openqa.suse.de/tests/9935878
https://openqa.suse.de/tests/9941441#next_previous
https://openqa.opensuse.org/tests/2868571 (opensuse aarch64)
https://openqa.suse.de/tests/9952666 (vnc-two-password@svirt-xen-hvm)
If we still have this kind of issue, we need to re-write on this test module.

### CODE REVIEW

File: tests/x11/vnc_two_passwords.pm [hunk 1]
```diff
@@ -132,15 +135,19 @@ sub run {
...
         generate_vnc_events $opt->{pw};
 
-        # Close xev
+        # Close xev, re-try if needed on aarch64, increase timeout on aarch64 for wait_serial, see poo#119416
+        my $timeout = (is_aarch64 && is_sle) ? 90 : 30;
         send_key 'ctrl-c';
-        wait_serial('xev-finished') || die 'xev not finished';
+        # give a second chance to quit xev
+        send_key 'ctrl-c' if (is_sle && is_aarch64) && ((check_screen 'need-to-close-again', $timeout) || (check_screen 'need-to-close-again-extra1', $timeout));
+        wait_serial('xev-finished', $timeout) || die 'performance issue on aarch64, see poo#120282 for more details';
```
Comment (rfan1) [inline tests/x11/vnc_two_passwords.pm:150]: Die messages are for all platforms, not only for aarch64, correct?

File: tests/x11/vnc_two_passwords.pm [hunk 2]
```diff
@@ -182,7 +189,9 @@ sub run {
         type_string("$display");
         send_key("ret");
         # We first test for a unsucessfull login
-        assert_screen('tigervnc-desktop-login');
+        # increase timeout on aarch64 for assert_screen, see poo#119416
+        my $timeout = (is_aarch64 && is_sle) ? 90 : 30;
```
Comment (rfan1) [inline tests/x11/vnc_two_passwords.pm:193]: I can see several functions use the same variable $timeout， can you please set it as global variable?

File: tests/x11/vnc_two_passwords.pm (hunk 1 above)
Comment (Zaoliang) [reply tests/x11/vnc_two_passwords.pm:150]: I checked, other plattforms don't have the issue at all, won't see this message, but this is good point, let me think how to limit this to aarch64 only, thanks!

File: tests/x11/vnc_two_passwords.pm (hunk 2 above)
Comment (Zaoliang) [reply tests/x11/vnc_two_passwords.pm]: yes, good idea. forgot to set this since there are some places need to be covered, thanks!

**Comment by Zaoliang** [comment]
checked again, looks good:
https://openqa.suse.de/tests/9974656

File: tests/x11/vnc_two_passwords.pm [hunk 3]
```diff
@@ -194,20 +203,34 @@ sub run {
...
         x11_start_program('vncviewer');
         send_key("ret");
-        assert_screen('tigervnc-desktop-login');
-        type_string("$options[1]->{pw}");
-        send_key("ret");
-        assert_screen('tigervnc-desktop-loggedin');
+        assert_screen([qw(tigervnc-desktop-login logged-as-priviled-user)]);
+        if (match_has_tag 'logged-as-priviled-user') {
+            send_key("alt-f4");
+            select_console('root');
```
Comment (rfan1) [inline tests/x11/vnc_two_passwords.pm:212]:
```suggestion
            select_console("root-console");
```

File: tests/x11/vnc_two_passwords.pm [hunk 4]
```diff
@@ -132,15 +136,20 @@ sub run {
...
         generate_vnc_events $opt->{pw};
 
-        # Close xev
+        # Close xev, re-try if needed on aarch64, increase timeout on aarch64 for wait_serial, see poo#119416
         send_key 'ctrl-c';
-        wait_serial('xev-finished') || die 'xev not finished';
+        # give a second chance to quit xev
+        send_key 'ctrl-c' if (is_sle && is_aarch64) && ((check_screen 'need-to-close-again', $timeout) || (check_screen 'need-to-close-again-extra1', $timeout));
+        my $message = 'xev not finished';
+        $message = 'xev not finished, performance issue on aarch64, see poo#120282 for more details' if (is_sle && is_aarch64);
```
Comment (rfan1) [inline tests/x11/vnc_two_passwords.pm:151]:
```suggestion
        $message .= ', performance issue on aarch64, see poo#120282 for more details' if (is_sle && is_aarch64);
```

File: tests/x11/vnc_two_passwords.pm [hunk 5]
```diff
@@ -194,20 +203,34 @@ sub run {
...
         # Test for a sucessfull login. Note: vncviewer remembers the last address, don't type it again
+        # sometimes screen is frozen with strange dialog like 'logged-as-priviled-user', there is no way to go further, so repeat these steps
         x11_start_program('vncviewer');
         send_key("ret");
-        assert_screen('tigervnc-desktop-login');
-        type_string("$options[1]->{pw}");
-        send_key("ret");
-        assert_screen('tigervnc-desktop-loggedin');
+        assert_screen([qw(tigervnc-desktop-login logged-as-priviled-user)]);
+        if (match_has_tag 'logged-as-priviled-user') {
```
Comment (rfan1) [inline tests/x11/vnc_two_passwords.pm:210]: Hi @Zaoliang,

I would suggest you add
`record_soft_failure('poo#120282');`

**Comment by Zaoliang** [comment]
https://openqa.suse.de/tests/9980819#next_previous shows pretty good results.

**Pull request merged**